doc/languages-frameworks/coq.section.md

```diff
@@ -88,3 +88,58 @@ with lib; mkCoqDerivation {
   };
 }
 ```
+
+## Three ways of overriding Coq packages {#coq-overriding-packages}
+
+There are three distinct ways of changing a Coq package by overriding one of its values: `.override`, `overrideCoqDerivation`, and `.overrideAttrs`.  This section explains what sort of values can be overridden with each of these methods.
+
+### `.override` {#coq-override}
+
+`.override` lets you change arguments to a Coq derivation.  In the case of the `multinomials` package above, `.override` would let you override arguments like `mkCoqDerivation`, `version`, `coq`, `mathcomp`, `mathcom-finmap`, etc.
+
+For example, assuming you have a special `mathcomp` dependency you want to use, here is how you could override the `mathcomp` dependency:
+
+```nix
+multinomials.override {
+  mathcomp = my-special-mathcomp;
+}
+```
+
+In Nixpkgs, all Coq derivations take a `version` argument.  This can be overridden in order to easily use a different version:
+
+```nix
+coqPackages.multinomials.override {
+  version = "1.5.1";
+}
+```
+
+Note that the derivation has to already know about the version you're trying to pass in.  For instance, the `multinomials` derivation above knows about the versions `1.5.2`, `1.5.1`, `1.5.0`, `1.5.0`, `1.4`, `1.3`, `1.2`, `1.1`, and `1.0`.  But if you tried to pass in version `2.0`, you would get a failure when you tried to build.  See the next section for how to add new versions.
```

Comment:
Here you should probably remind that the `version` argument to a coq derivation has special support for a whole range of additional things beyond known versions. Quoting from the doc above:

>* if it is a majorMinor `"x.y"` prefix of a known released version (as defined above), then the latest `"x.y.z"` known released version is selected (for the ordering given by `versionAtLeast`),
>* if it is a path or a string representing an absolute path (i.e. starting with `"/"`), the provided path is selected as a source, and the `version` attribute of the resulting derivation is set to `"dev"`,
>* if it is a string of the form `owner:branch` then it tries to download the `branch` of owner `owner` for a project of the same name using the same vcs, and the `version` attribute of the resulting derivation is set to `"dev"`, additionally if the owner is not provided (i.e. if the `owner:` prefix is missing), it defaults to the original owner of the package (see below),
>* if it is a string of the form `"#N"`, and the domain is github, then it tries to download the current head of the pull request `#N` from github,

Reply:
This kind of stuff should definitely be noted.

Reply:
Good call.

I added a little note in b24c9093106 that tells the user to refer to the above section for the different values that can be passed to `version`.

Reply:
The additional sentence does not suggest that the `version` attribute also supports building versions that it doesn't know about. I suggest editing one of the previous sentences ("Note that the derivation has to already know about the version you're trying to pass in.") to convey this piece of information.

Reply:
In a2a287e9373 I just took out the note about the restrictions on `version`, and instead just left in the sentence pointing the user to the docs for what works as a valid `version`.  I felt it was a little overkill trying to re-explain how `version` works in this section, when it was really just an overview of how to use `.override`.

How does this look?

Reply:
LGTM, this is a reasonable choice I think.

Reply:
I'd suggest squashing before merging or squash-merging.

Reply:
Thanks, I squashed into 6cc6850fe23.